numpy/lib/function_base.py

```diff
@@ -2286,6 +2286,25 @@ class vectorize:
     """
     def __init__(self, pyfunc=np._NoValue, otypes=None, doc=None,
                  excluded=None, cache=False, signature=None):
+
+        if not callable(pyfunc):
+            p_temp = pyfunc
+            pyfunc = np._NoValue
+            if p_temp is not None and p_temp is not np._NoValue:
+                o_temp = otypes
+                otypes = p_temp
```

Comment:
What if someone has given really wrong input, like `vectorize(1, otypes=[...])`. I fear that for the shifting, one needs to do a `*args, **kwargs` and re-call (or re-init) with `None, *args, **kwargs` so that python can remove duplications...

Reply:
>What if someone has given really wrong input

That would be assuming that the user is trying to use vectorize without reading the documentation, which I don't think is right. Anyways, from tinkering with the code while trying to solve this, it seems that what would happen is that vectorize would raise ```ValueError("Invalid otype specification")```. There might be some specific incorrect input that avoids this, but I don't think that's something we have to be concerned about.